// Copyright (c) 2025, Alliance for Open Media. All rights reserved
//
// This source code is subject to the terms of the BSD 3-Clause Clear License
// and the Alliance for Open Media Patent License 1.0. If the BSD 3-Clause Clear
// License was not distributed with this source code in the LICENSE file, you
// can obtain it at www.aomedia.org/license/software-license/bsd-3-c-c. If the
// Alliance for Open Media Patent License 1.0 was not distributed with this
// source code in the PATENTS file, you can obtain it at
// www.aomedia.org/license/patent.

edition = "2023";

package iamf_tools_cli_proto;

enum OutputAudioFormat {
  option features.enum_type = CLOSED;

  OUTPUT_FORMAT_INVALID = 0;

  // Omit output files.
  OUTPUT_FORMAT_NONE = 1;

  // Wav file output. Bit depth is automatically determined based on the
  // underlying codec.
  OUTPUT_FORMAT_WAV_BIT_DEPTH_AUTOMATIC = 2;

  // Wav file output, with explicit bit depths.
  OUTPUT_FORMAT_WAV_BIT_DEPTH_SIXTEEN = 3;
  OUTPUT_FORMAT_WAV_BIT_DEPTH_TWENTY_FOUR = 4;
  OUTPUT_FORMAT_WAV_BIT_DEPTH_THIRTY_TWO = 5;
}
